src/v/cloud_topics/housekeeper/tests/housekeeper_test.cc

```diff
@@ -32,6 +32,14 @@ class fake_l0_metastore
       : _tidp(tidp)
       , _start_offset(start_offset) {}
 
+    kafka::offset
+    get_start_offset(const model::topic_id_partition& tidp) override {
+        if (tidp == _tidp) {
+            return _start_offset;
+        }
+        return kafka::offset{0};
```

Comment:
nit: not a great practice but if you're set on using kafka::offset instead of a std::optional, it isn't uncommon to have "unset" be kafka::offset{}, which defaults to the min val